### DISCUSSION TITLE
Making debugging flow easier

### DISCUSSION THREAD

**Discussion by kannanvijayan** [RecordReplay/devtools#2854] [Ideas]
These two thoughts came from my first "real" session with replay trying to track down an issue with the video player code.

1. Logpoints are too disconnected from each other.  I can't get all of my logpoints to show up on the timeline.  I found myself using the print-eval command in the logpoints and looking at the console for when things executed, but I was forced to name each of the logpoints with unique strings to keep track of things.
I'd really like to be able to associate different colors/icons with different logpoints, and have some way of making them all show up in the timeline so I can visually distinguish "when things are happening".  It'd be cool if there was a drop-down arrow on each logpoint where I could associate it with a unique dot (either color, or a little panda face, or a little dog face, different icon for each one).. and then have those show up on the timeline.

2. I really disliked having to switch back and forth from comments view to the pause-points view to be able to use the debugger step/continue/rewind/etc. functions.  It broke up the flow.  I found myself wishing for some way to make those controls visible at all times, separate from the rest of the UI changing.

**Comment by jonbell-lot23** [comment]
Just a note that this is a good point and we're seeing similar feedback elsewhere. To discuss and consider!

**Comment by colevandersWands** [comment]
> 2. I really disliked having to switch back and forth from comments view to the pause-points view ...

I came across this as well.  Maybe a "Comments" drop-down in the debugger would help, it could show all comments associated with that step of execution.   The UI could be simple and embedded with the current debugger.

When the panel is open it could give you the option to create a new comment connected to either:
- the current line, every time it's executed
- the current step

![comments-section-in-debugger](https://user-images.githubusercontent.com/18554853/137945409-accb0ac7-b955-40fd-a409-d00f31fff657.png)

**Comment by jasonLaster** [comment]
This reminds me of these two issues as well

https://github.com/RecordReplay/devtools/issues/3869
https://github.com/RecordReplay/devtools/issues/3644

There are probably a couple of things going on here:

1. it would be nice to see comments and scopes sometimes. The answer here is probably more flexible layouts which we've discussed in the past.
2. it would be nice to easily be able to add a comment on a line. Partial credit would be a context menu in the gutter and/or a global add comment button in the comments pane.

**Reply by colevandersWands** [reply]
flexible layouts sounds interesting, is there a best place to catch up on the history of this discussion?

**Reply by jasonLaster** [reply]
@colevandersWands that's tough to find, but also mostly historical too

**Reply by jonbell-lot23** [reply]
Thanks for your comments @colevandersWands ! On the design side, we don't have a nicely packaged thread or screen to point you to, but it's more of a "discussions that keep coming up" sort of fidelity.

I'd love to ask a question for some design research: have you ever discovered the way to hide the video player or editor? No worries if not, it's pretty hidden. We know as users of devtools/IDEs/professional software that *of course* we need some flexibility in the UI, and the "hide video" and "hide editor" buttons are a baby step towards that.

**Reply by colevandersWands** [reply]
I hadn't looked for it but had minimized the video as much as possible to get i out of the way.  after you said there was a way I found it quickly, the UI isn't cluttered and it was where I thought it might be after I knew to look.

**Reply by jonbell-lot23** [reply]
Thanks very much! Good to hear your experience, and I'm eager to make this more helpful and discoverable in future UI updates.

**Reply by jonbell-lot23** [reply]
Hello @colevandersWands, we recently shipped a feature that you might enjoy:

<img width="252" alt="image" src="https://user-images.githubusercontent.com/9154902/158482458-6c722d34-7111-4569-a58f-25b14728e51e.png">

This helps a lot with real estate on the screen. Curious to hear your thoughts!

**Reply by colevandersWands** [reply]
Thanks for the mention @jonbell-lot23, this feature is nice.  Much easier to focus on what's important in a moment.

Funny story though. I wasn't able to test it out on an older replay I have because of an error loading it, so I got to try it out on [a replay of an error loading a replay](https://app.replay.io/recording/error-when-loading-older-replay--2c0a4628-9596-4fa9-ba46-4f7daf27ab8f) 😵‍💫  (I've invited you with the email from your website)
